src/platform/OpenThread/GenericThreadStackManagerImpl_OpenThread.cpp

```diff
@@ -2647,32 +2690,56 @@ void GenericThreadStackManagerImpl_OpenThread<ImplClass>::OnDnsBrowseResult(otEr
         serviceInfo.mTxtData            = txtBuffer;
         serviceInfo.mTxtDataSize        = sizeof(txtBuffer);
 
-        error = MapOpenThreadError(otDnsBrowseResponseGetServiceInfo(aResponse, serviceName, &serviceInfo));
+        otError err = otDnsBrowseResponseGetServiceInfo(aResponse, serviceName, &serviceInfo);
+        error       = MapOpenThreadError(err);
 
-        VerifyOrExit(error == CHIP_NO_ERROR, );
+        VerifyOrExit(err == OT_ERROR_NOT_FOUND || err == OT_ERROR_NONE, );
 
         DnsResult * dnsResult = Platform::New<DnsResult>(aContext, CHIP_NO_ERROR);
-        error = FromOtDnsResponseToMdnsData(serviceInfo, type, dnsResult->mMdnsService, dnsResult->mServiceTxtEntry);
-        if (CHIP_NO_ERROR == error)
+
+        VerifyOrExit(dnsResult != nullptr, error = CHIP_ERROR_NO_MEMORY);
+
+        // If SRV, TXT or AAAA record was not included, send additional DNS query using DNS resolve
+        if (err == OT_ERROR_NOT_FOUND || serviceInfo.mTxtDataSize == 0 || otIp6IsAddressUnspecified(&serviceInfo.mHostAddress))
         {
-            // Invoke callback for every service one by one instead of for the whole
-            // list due to large memory size needed to allocate on stack.
-            static_assert(ArraySize(dnsResult->mMdnsService.mName) >= ArraySize(serviceName),
-                          "The target buffer must be big enough");
-            Platform::CopyString(dnsResult->mMdnsService.mName, serviceName);
-            DeviceLayer::PlatformMgr().ScheduleWork(DispatchBrowse, reinterpret_cast<intptr_t>(dnsResult));
+            const otDnsQueryConfig * defaultConfig = otDnsClientGetDefaultConfig(ThreadStackMgrImpl().OTInstance());
+
+            error = MapOpenThreadError(otDnsClientResolveService(ThreadStackMgrImpl().OTInstance(), serviceName, type,
+                                                                 OnDnsBrowseInvokedResolveResult,
+                                                                 reinterpret_cast<void *>(dnsResult), defaultConfig));
```

Comment:
Same here. We can pass `NULL` instead of `defaultConfig` which will then use default config.